api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,388 @@ paths:
+        The behavior of commit is specific to `commit_mode` and `commit_semantics` parameters of the session
+        that makes a commit. Please see details in subscription streaming endpoint specification.
+      parameters:
+        - name: subscription_id
+          in: path
+          type: string
+          description: Id of subscription
+          required: true
+        - name: X-Nakadi-SessionId
           in: header
+          type: string
+          description: |
+            Id of a session (client) which makes a commit. It is not possible to make a commit from a terminated or
+            none-existing session. Also the session can't commit something which was not sent to it.
+          required: true
+        - name: cursors
+          in: body
+          description: Cursors with offsets to commit.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Cursor'
+          required: true
+      responses:
+        '200':
+          description: Offsets were committed
+        '204':
+          description: |
+            No Content. At least one cursor which was tried to be committed is older or equal to already committed one.
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Client is not authorized to perform this operation
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Service (temporarily) unavailable
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions/{subscription_id}/events:
+    get:
+      tags:
+        - stream-api
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Starts a new stream for reading events from this subscription. The data will be automatically rebalanced
+        between streams of one subscription. The minimal consumption unit is a partition, so it is possible to create as
+        many streams as the total number of partitions in event-types of this subscription. The rebalance currently
+        only operates with the number of partitions so the amount of data in event-types/partitions is not considered
+        during autorebalance.
+        The position of the consumption is managed by Nakadi. The client is required to commit the cursors he gets in
+        a stream. For details of how committing works see description of parameters `commit_mode`, `commit_semantics`
+        and the description of commit endpoint.
```

Comment:
yes, that's right -- i would prefer `commit_each` as default behaviour (or even as only behaviour, entirely removing `commit_mode` parameter). i think that it best fits the linear nakadi processing model.

can you give an example where `commit_at_will` actually makes things easier for the consumer? i think it will just mask / encourage "messy" processing.